nmi_mysql/nmi_mysql.py

```diff
@@ -11,7 +11,31 @@
 
 class DB(object):
 
-    def __init__(self, conf, autoconnect=False):
+    def __init__(self, conf, max_pool_size=20):
+        self.conf = conf
+        self.max_pool_size = max_pool_size
+        self._initialize_pool()
+
+    def _initialize_pool(self):
+        self.pool = Queue(maxsize=self.max_pool_size)
+
+        for _ in range(0, self.max_pool_size):
+            self.pool.put_nowait(Connection(self.conf))
+
+    def connect(self):
+        con = self.pool.get(True)
+        con.connect()
+
+        return con
+
+    def close(self, con):
+        con.close()
+        self.pool.put_nowait(con)
```

Comment:
I updated the implementation.
- When instantiating `DB`, it will create a connection pool with all `Connection` instances not yet connected.
- When executing a query, it will get a `Connection` instance from the pool, connect to the database, execute the query, close the connection, and return the `Connection` instance to the pool.